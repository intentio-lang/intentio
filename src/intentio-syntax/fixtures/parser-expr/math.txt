{
    3 + 4 * 5 - ( 8 + 1) / 3;
    +3;
    -(2+2);
    +foo();
    -2 and bar();
    not x and b or not c;
    2 == 1 + 1;
# FIXME:     3 /= 1 + 1 and 4 < 9;
    6 * 5 >= 0 and not 2 <= 9;
}
