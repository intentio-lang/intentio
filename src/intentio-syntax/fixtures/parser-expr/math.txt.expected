BlockExpr 
    ( Block 
        [ BinExpr BinSub ( BinExpr BinAdd ( LitExpr ( Integer "3" ) ) ( BinExpr BinMul ( LitExpr ( Integer "4" ) ) ( LitExpr ( Integer "5" ) ) ) ) ( BinExpr BinDiv ( ParenExpr ( BinExpr BinAdd ( LitExpr ( Integer "8" ) ) ( LitExpr ( Integer "1" ) ) ) ) ( LitExpr ( Integer "3" ) ) )
        , UnaryExpr UnaryAdd ( LitExpr ( Integer "3" ) )
        , UnaryExpr UnarySub ( ParenExpr ( BinExpr BinAdd ( LitExpr ( Integer "2" ) ) ( LitExpr ( Integer "2" ) ) ) )
        , UnaryExpr UnaryAdd ( FunCallExpr ( IdExpr ( Id ( ScopeId "foo" ) ) ) ( FunArgs [] ) )
        , BinExpr BinAnd ( UnaryExpr UnarySub ( LitExpr ( Integer "2" ) ) ) ( FunCallExpr ( IdExpr ( Id ( ScopeId "bar" ) ) ) ( FunArgs [] ) )
        , BinExpr BinOr ( BinExpr BinAnd ( UnaryExpr UnaryNot ( IdExpr ( Id ( ScopeId "x" ) ) ) ) ( IdExpr ( Id ( ScopeId "b" ) ) ) ) ( UnaryExpr UnaryNot ( IdExpr ( Id ( ScopeId "c" ) ) ) )
        , BinExpr BinEqEq ( LitExpr ( Integer "2" ) ) ( BinExpr BinAdd ( LitExpr ( Integer "1" ) ) ( LitExpr ( Integer "1" ) ) )
        , BinExpr BinAnd ( BinExpr BinGtEq ( BinExpr BinMul ( LitExpr ( Integer "6" ) ) ( LitExpr ( Integer "5" ) ) ) ( LitExpr ( Integer "0" ) ) ) ( BinExpr BinLtEq ( UnaryExpr UnaryNot ( LitExpr ( Integer "2" ) ) ) ( LitExpr ( Integer "9" ) ) )
        ] 
    )